Assets/Source/Player/Scripting/Interface/App/SystemJsApi.cs

```
@@ -200,6 +212,40 @@ public void restart()
 #else
             Log.Warning(this, "Restart not supported for this platform.");
 #endif
-        }        
+        }
+
+        /// <summary>
+        /// Starts the application log-in flow.
+        /// </summary>
+        /// <param name="force">If true, forces the user to log out</param>
+        public void login(bool force=false)
+        {
+            var creds = _config.Network.Credentials;
+            if (creds.IsLoggedIn() && !creds.IsGuest())
+            {
+                if (force)
+                {
+                    Log.Info(this, "Forcing log out.");
+                    _messages.Publish(MessageTypes.SIGNOUT);
+                }
+                else
+                {
+                    Log.Info(this, "User is already logged in.");
+                }
+                return;
+            }
+
+            Log.Info(this, "Starting user log in.");
+            _messages.Publish(MessageTypes.LOGIN);
+        }
+
+        /// <summary>
+        /// Logs out a user.
+        /// </summary>
+        public void logout()
+        {
+            Log.Info(this, "Logging out user");
+            _messages.Publish(MessageTypes.SIGNOUT);
+        }

```

This is looking great. I do think we should alter the interface slightly: not a huge fan of the `bool` parameter. If I read `login(true)`, I would not think "yah this is going to log me out." How about:

```
login()
logout()
isLoggedIn
```

A cleaner interface that can handle edge cases well (`login` and `logout` can silently do nothing).

---

agreed 👍